integration/exception-handling/src/main/java/com/baumeister/sndbx/exceptionhandling/service/exception/ProblemDetailsMapper.java

```diff
@@ -0,0 +1,80 @@
+package com.baumeister.sndbx.exceptionhandling.service.exception;
+
+import com.baumeister.sndbx.exceptionhandling.general.exception.NotFoundException;
+import com.baumeister.sndbx.exceptionhandling.general.exception.ValidationException;
+import com.baumeister.sndbx.exceptionhandling.service.exception.ProblemDetailsFactoryHelper.ProblemDetailsFactoryHelperBuilder;
+import com.devonfw.devon4j.generated.api.model.ValidationProblemDetailsTo;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+import org.mapstruct.factory.Mappers;
+import com.devonfw.devon4j.generated.api.model.ProblemDetailsTo;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+
+public class ProblemDetailsMapper {
+
+  private static ProblemDetailsMapper instance;
+
+  private ProblemDetailsMapper() {
+  }
+
+  public static ProblemDetailsMapper getInstance() {
```

Comment:
synchronized?

Reply:
@Synchronized would block a parallel access. Therefore, I would not do it here. In worst case two calls are reaching the instance at the same time and one overwrites the instance with a new ProblemDetailsMapper, which is not leading to a problem. The only downside would be an unnecessary double instantiation. As this can only be the case, on the very first call of getInstance(), I don't think this adds much value.

Reply:
You should add a comment that this is on purpose. Otherwise, it looks like you forgot it. Or you can have a look at https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java. But this is a bit painful Java.